Add DotMenu component

ash3rz opened the pull request.

ash3rz committed:
Add DotMenu component

ash3rz committed:
Stop event propagation in DotMenu clicks

psarando commented on stories/DotMenu.stories.js:
```diff
@@ -0,0 +1,48 @@
+    }
+}
+
+storiesOf("DotMenu", module).add("with sample menu items", () => (
```
Does the `ui-lib` support the new style `Component Story Format` that are simple exports now?

ash3rz replied:
Ah, good catch.  I'll push a change to update to the new format.  It requires a slight change to the storybook config, but it doesn't break the stories in the old format.

psarando commented on src/components/dotMenu/DotMenu.js:
```diff
@@ -0,0 +1,76 @@
+ *  MenuItems should be passed into the render prop which in turn
+ *  will provide a function for closing the menu.
+ *
+ *  Sample usage:
```
Good comments 👍

psarando commented on src/components/dotMenu/DotMenu.js:
```diff
@@ -0,0 +1,76 @@
+import build from "../../util/DebugIDUtil";
+
+function DotMenu(props) {
+    const { baseId, render, ButtonProps, MenuProps } = props;
```
👍

psarando reviewed (approved):
LGTM :shipit:

ash3rz committed:
Simplify storybook config

This works for stories in the old format using `storiesOf`, but will also now work with stories written in storybook's new CSF way

ash3rz committed:
Refactor DotMenu story into Component Story Format

ash3rz commented:
Thanks for the review!
